lib/hsa/hsaTrackingUpdatePhase.cpp

```diff
@@ -68,22 +68,41 @@ hsaTrackingUpdatePhase::hsaTrackingUpdatePhase(Config& config, const std::string
     // Gain stuff here
     gain_len = 2 * 2048 * _num_beams * sizeof(float);
     host_gain = (float*)hsa_host_malloc(gain_len, device.get_gpu_numa_node());
+    if (host_gain == nullptr)
+        throw std::runtime_error("Could not allocate memory in hsaTrackingUpdatePhase");
+
     gain_buf = host_buffers.get_buffer("gain_tracking_buf");
     register_consumer(gain_buf, unique_name.c_str());
     gain_buf_id = 0;
+    if (gain_len != gain_buf->frame_size)
+        throw std::runtime_error("The gain_len in does not match the buffer frame size");
 
-    // Phase here
+    // Phase and scaling data go into the same array so they can be transferred at the same time
+    // but are referenced by two pointers to make it easier to index.
     phase_frame_len = _num_elements * _num_beams * 2 * sizeof(float);
+    scaling_frame_len = _num_beams * sizeof(float);
     // Two alternating banks
-    host_phase_0 = (float*)hsa_host_malloc(phase_frame_len, device.get_gpu_numa_node());
-    host_phase_1 = (float*)hsa_host_malloc(phase_frame_len, device.get_gpu_numa_node());
+    host_phase_0 =
+        (float*)hsa_host_malloc(phase_frame_len + scaling_frame_len, device.get_gpu_numa_node());
+    if (host_phase_0 == nullptr)
+        throw std::runtime_error("Could not allocate memory in hsaTrackingUpdatePhase");
+    host_scaling_0 = host_phase_0 + _num_elements * _num_beams * 2;
+
+    host_phase_1 =
+        (float*)hsa_host_malloc(phase_frame_len + scaling_frame_len, device.get_gpu_numa_node());
+    if (host_phase_1 == nullptr)
+        throw std::runtime_error("Could not allocate memory in hsaTrackingUpdatePhase");
+    host_scaling_1 = host_phase_0 + _num_elements * _num_beams * 2;
```

Comment:
shouldn't this be:
```
host_scaling_1 = host_phase_1 + _num_elements * _num_beams * 2;
```

Reply:
You are right, good catch!  That would have been a hard bug to find.